packages/docs/docsite/src/components/splash/splash.tsx

```diff
@@ -0,0 +1,357 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+
+import React, {
+	useEffect,
+	useState,
+	useRef,
+	RefObject,
+	useMemo,
+	useCallback,
+} from 'react'
+import anime from 'animejs'
+import { randomBetween } from '../../util/util'
+import styled from 'styled-components'
+import Footer, { FOOTER_HEIGHT } from '../footer'
+import * as HeaderComp from '../header/header'
+import { useWindowDimensions } from '../../util/hooks/useWindowDimensions'
+import LogoBG from './logobg'
+import { TimelineMax, Elastic } from 'gsap'
+
+const HeaderComponent = HeaderComp.Header
+const headerHeight = HeaderComp.HEIGHT
+
+// use different layout for click divs based on screen size
+const minScreenSize = '768px'
+
+const IndexPage = () => {
+	const [animatingOut, setAnimationOut] = useState<boolean>(false)
+	const titleRef = useRef<RefObject<any> | null | undefined>(null)
+	const docRef = useRef<RefObject<any> | null | undefined>(null)
+	const blogRef = useRef<RefObject<any> | null | undefined>(null)
+	const sourceRef = useRef<RefObject<any> | null | undefined>(null)
+
+	const { height, width } = useWindowDimensions()
+
+	useEffect(() => {
```

Comment:
This animation is large enough that I think it warrants its own custom hook function, possibly in a separate file.

maybe `useSplashPageMountAnimation()`